inputgraph/loader.go

```diff
@@ -397,8 +404,13 @@ func (l *loader) handleFor(ctx context.Context, forStmt spec.ForStatement) error
 }
 
 func flattenForArgs(args []string, seps string) []string {
+	// The first 2 args will be the index variable name and IN so we ignore the
+	// first 2 values. Example: "FOR idx IN vars..." etc.
+	if len(args) < 3 {
+		return nil
+	}
 	var ret []string
-	for _, arg := range args {
+	for _, arg := range args[2:] {
```

Comment:
does this mean changing `FOR foo IN ...` to `FOR bar IN` has no effect on the hash, or is it hashed elsewhere?

If it doesn't affect the hash, this seems wrong as it could change the behaviour of the command that's being executed.

Reply:
The command's args are hashed beforehand. https://github.com/earthly/earthly/blob/mh%2Fauto-skip-if-exists/inputgraph/loader.go#L274

Reply:
ah perfect!